upt_macports/upt_macports.py

```diff
@@ -173,9 +178,38 @@ def create_package(self, upt_pkg, output=None):
         }
 
         try:
+            self.frontend = upt_pkg.frontend
             pkg_cls = pkg_classes[upt_pkg.frontend]
         except KeyError:
             raise upt.UnhandledFrontendError(self.name, upt_pkg.frontend)
-
         packager = pkg_cls()
         packager.create_package(upt_pkg, output)
+
+    def package_version(self, name):
+        pkg_classes = {
+            'pypi': MacPortsPythonPackage,
+            'cpan': MacPortsPerlPackage,
+            'rubygems': MacPortsRubyPackage,
+            'npm': MacPortsNpmPackage
+        }
+        try:
+            port_name = pkg_classes[self.frontend](
+            )._normalized_macports_name(name)
+        except KeyError:
+            raise upt.UnhandledFrontendError(self.name, self.upt_pkg.frontend)
+
+        self.logger.info(f'Checking MacPorts tree for port {port_name}')
+        cmd = f'port info --version {port_name}'
+        port = subprocess.getoutput(cmd)
+        # TODO: version not found when macports tree is not updated
```

Comment:
I don't understand this TODO message. If the MacPorts tree is not up-to-date there are two options: (i) the port doesn't exist yet in the local tree but is available in the GitHub tree or (ii) the version that will be returned is lower if the port has been updated already in the GitHub tree. So I don't think there is anything else left TODO here, or... what are you thinking about here?

Reply:
So earlier when I was testing recursive packaging, it stopped working without any changes in code which was dues to change in output of `port info` as MacPorts tree was not updated so the output was:
```
Warning: "Update message"
version: 0.1.2
```
because of this I had added the TODO which still is not handled properly.
We can either use the local version or sys.exit if such Warning is there.

Reply:
Now this is solved by changing second condition to
```
 elif 'version' in port:
            curr_ver = port.split('version: ')[1]
```

Reply:
ah, good point - I guess I never go two weeks without updating the port definitions ;) Then just add another ```elif``` statement to catch this and emit a proper message right now instead of adding a TODO statement.

Reply:
> Now this is solved by changing second condition to
> 
> ```
>  elif 'version' in port:
>             curr_ver = port.split('version: ')[1]
> ```

I don't think that is a good idea as the MacPorts warning of having a >2 weeks outdated ports tree will get buried in this case. As I said, I would handle this as a separate condition and when we run into this situation warn the user about his/her outdated ports tree and tell them to resolve that first (i.e., to a ```sys.exit``` here). Many ports will have been updated within this two week period and working with an outdated tree is never a good idea when trying to package stuff. In addition, we should add in the (still to be written) documentation and/or README that one is assumed to have an up-to-date, local MacPorts tree.

Reply:
In general use I agree with not proceeding if the ports are outdated, however there are times when that's appropriate (like trying to keep a system at a static set of ports but you need to add another).  If we can include an override (maybe -f is good enough?) so that it is possible to run this on an "outdated" tree I think that'd solve the use case.

Reply:
thanks for the feedback @ghosthound. 

Adding an override flag would likely need to be done in ```upt``` so that would require some additional considerations (e.g., is this a general enough use-case for other backends or something specific to MacPorts, are there more things that one could/should override with the ```-f``` flag, etcetera).

I am not necessarily set on doing a ```sys.exit``` and *requiring* the user to update the ports tree (although this would be in general the preferred action). We could consider to clearly log that "the tree is outdated and we recommend to update the tree first", and then leave it to the user to decide. But for sure, I would rather not ignore the warning message of an outdated tree as currently happens.